flexmeasures/data/services/forecasting.py

```diff
@@ -244,31 +233,14 @@ def make_rolling_viewpoint_forecasts(
     )
     click.echo("Job %s made %d forecasts." % (rq_job.id, len(forecasts)))
 
-    ts_value_forecasts = [
-        timed_value_type(
-            use_legacy_kwargs=False,
+    for dt, value in forecasts.items():
+        TimedBelief(
             event_start=dt,
             belief_horizon=horizon,
             event_value=value,
             sensor=sensor,
             source=data_source,
         )
```

Comment:
I should note that I was surprised to find out at this point that I no longer had to explicitly save the TimedBelief objects to the database. I had first replaced the code below with a call to `api.common.utils.api_utils.save_to_db`, which works well for BeliefsDataFrames, using:
```
bdf = tb.BeliefsDataFrame(ts_value_forecasts)
save_to_db(bdf)
```
That worked because `save_to_db` checks for unchanged beliefs before attempting to save. However, the original code using `save_to_session` resulted in duplicate key value errors, which, as I found out, was due to the initialization calls to TimedBelief already resulting in the objects being saved. Subsequently, I made [a commit](https://github.com/FlexMeasures/flexmeasures/pull/287/commits/fb58ec7459dd69c571bee27cdce61e67c14617ae) to forego of the redundant calls to `save_to_db`.

If you have any thoughts about why this is happening, please let me know. As it stands, is this code going against your earlier work on API calls resulting in a single db transaction?